### PR TITLE
Implement Redis-NATS proxy server with TCP connection pooling

### PR DESCRIPTION
**Pull request by igorrius**
This implements a Redis-NATS proxy server with a TCP connection pooling feature that allows reusing of active connections for multiple requests rather than opening a new connection for each request. This is achieved by creating a 'NetConnPullManager' and new NATS connection management structures.

### CODE REVIEW

**Comment by inimbir** [comment]
Cool!

File: example/client/main.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		slog.Error("connect to nats", "err", err)
+		return
+	}
+	defer nc.Close()
+
+	// read first argument as the address of the Redis server
+	// if no argument is provided, use the default address
+	addr := "localhost:6379"
+	if len(os.Args) > 1 {
+		addr = os.Args[1]
```
Comment (inimbir) [inline example/client/main.go:35]: Seems we never check how many args we have

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module github.com/Autodoc-Technology/redis-nats-proxy
+
+go 1.22.0
+
+require (
+	github.com/nats-io/nats.go v1.34.0
```
Comment (inimbir) [inline go.mod:6]: Nats  :))))

**Review by inimbir — APPROVED** [review]

File: example/client/main.go (hunk 1 above)
Comment (igorrius) [reply example/client/main.go:35]: ugu

**Pull request merged**